packages/calcite-components/src/components/list/list.tsx

```diff
@@ -235,14 +246,22 @@ export class List
    *
    * @readonly
    */
-  @property() filteredData: ItemData = [];
+  @property() filteredData: (ItemData | ItemGroupData)[] = [];
+
+  /**
+   * The currently filtered `calcite-list-item's` & `calcite-list-item-group's`.
+   *
+   * @readonly
+   * @deprecated Use `filteredResults` instead.
+   */
+  @property() filteredItems: ListElement[] = [];
```

Comment:
since it is a short notice, its better to deprecate first. To avoid confusing with naming, `filteredResults` is added.

If we are including groups in filter, the user should be able to access the groups me thinks.

Reply:
Part of my reasoning for not including groups is that they aren't results. They can't be selected, they don't have a value, etc.

Reply:
Agree item-group is more of a  presentational component, but using `heading` as `filterProp` can affect how filtering is done. And users should be able to lookup the API for any potential matches for filterText.

Did think of adding direct children items of the group as filteredItems instead of the item-group but that can be misleading and not consistent with how filteredItems are added when item is nested inside a item without a group.

Reply:
It will make it easier if we don't include `item-group` in filteredItems but filteredItems can't be empty if there is a match.

Did think of adding direct children items of the item-group as filteredItems instead of the item-group but that can be misleading and not consistent with how items are added to `filteredItems` when item is nested inside a item without a group. Thoughts?

Reply:
After reviewing the issue again, I _think_ the request was primarily visual and not related to `filteredItems`. My earlier context was minimal, so apologies for any confusion. 😅

I agree on including them when they are valueless is a bit odd. 

Could we proceed with the behavior while keeping `filteredItems` as is?